spacify-portal/src/main/java/org/uci/spacifyPortal/services/RoomService.java

```diff
@@ -0,0 +1,40 @@
+package org.uci.spacifyPortal.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.uci.spacifyLib.entity.RoomEntity;
+import org.uci.spacifyLib.entity.RoomType;
+import org.uci.spacifyLib.repsitory.RoomRepository;
+import org.uci.spacifyPortal.utilities.CreateRequest;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class RoomService {
+    @Autowired
+    private RoomRepository roomRepository;
+
+    public RoomEntity createRoom(CreateRequest createRequest) {
+        RoomEntity roomEntity = new RoomEntity();
+        roomEntity.setRoomRules(null);
+        roomEntity.setTippersSpaceId(createRequest.getTippersSpaceId());
+        // we set roomId to -1, as the database trigger: setDefault will generate the next ID and use that for the insert.
+        roomEntity.setRoomId(-1L);
+        roomEntity.setRoomType(RoomType.valueOf(createRequest.getRoomType()));
+        roomEntity.setDescription(createRequest.getRoomDescription());
```

Comment:
RoomEntity does not have a description field. This would give an exception when stored. Do we have this field in the ER diagram?